components/brave_wallet/resources/solana_provider.js

```diff
@@ -31,4 +31,26 @@
       writable: false
     }
   })
+  Object.defineProperties(window.solana, {
+    on: {
+      value: SolanaEventEmitter.on,
+      writable: false
+    },
+    off: {
+      value: SolanaEventEmitter.off,
+      writable: false
+    },
+    emit: {
+      value: SolanaEventEmitter.emit,
+      writable: false
+    },
+    removeListener: {
+      value: SolanaEventEmitter.removeListener,
+      writable: false
+    },
+    removeAllListeners: {
+      value: SolanaEventEmitter.removeAllListeners,
+      writable: false
+    }
```

Comment:
Please don't revert the second commit
https://github.com/brave/brave-core/pull/14282/commits/caad42865875834f3019317a96e95eb468957309
It will cause console error